pandevice/firewall.py

```diff
@@ -360,6 +361,64 @@ def commit_policy_and_objects(self, sync=False, exception=False):
         return self._commit(sync=sync, exclude="policy-and-objects",
                             exception=exception)
 
+    def organize_into_vsys(self, create_vsys_objects=True, refresh_vsys=True):
+        """Organizes all imported objects under the appropriate Vsys object.
+
+        Args:
+            create_vsys_objects (bool): Create the vsys objects (True) or use the ones already connected to this firewall (False).
+            refresh_vsys (bool): Refresh all vsys objects' parameters before doing the reorganization or not.  This is assumed True if create_vsys_objects is True.
+
+        """
+        from pandevice import device
+        from pandevice import network
+
+        # Mapping of device.Vsys params to pandevice classes.
+        mapping = {
+            'interface': network.Interface,
+            'vlans': network.Vlan,
+            'virtual_wires': network.VirtualWire,
+            'virtual_routers': network.VirtualRouter,
+        }
+
+        # Optional: create the vsys objects.
+        if create_vsys_objects:
+            device.Vsys.refreshall(self, name_only=True)
+
+        # Vsys to put objects into.
+        available_vsys = [x for x in self.children
+                          if isinstance(x, device.Vsys)]
+
+        # Optional: refresh the vsys params.
+        if create_vsys_objects or refresh_vsys:
+            for x in available_vsys:
+                x.refresh(refresh_children=False)
```

Comment:
Since this part wasn't a bottleneck, I leaned away from the single API call.  The "only works with candidate config" bit was part of why I abandoned that, plus parsing the returned XML would require more speciality code.